File: model/SysFile.py
from datetime import datetime

from sqlalchemy import Column, String, BigInteger, Text, DateTime, SmallInteger, Integer

from base import db

class SysFile(db.Model):
    __tablename__ = 'sys_file'

    id = Column(Integer, primary_key=True, autoincrement=True, comment='文件id')
    type = Column(String(100), nullable=True, comment='文件类型')
    url = Column(String(255), nullable=True, comment='文件url')
    name = Column(String(100), nullable=False, comment='文件原始名')
    object_name = Column(String(100), nullable=False, comment='文件存储名')
    size = Column(String(100), nullable=False, default='0', comment='文件大小')
    path = Column(String(255), nullable=False, comment='文件路径')
    md5 = Column(String(32), unique=True, nullable=False, comment='文件的MD5值，用于比对文件是否相同')
    create_time = Column(DateTime, nullable=False, default=datetime.utcnow, comment='创建时间')
    update_time = Column(DateTime, nullable=True, default=datetime.utcnow, onupdate=datetime.utcnow, comment='更新时间')

File: dehaze-go/model/sys_file.go
package model

// SysFile 文件表
type SysFile struct {
	BaseModel
	Type       string `gorm:"column:type;type:varchar(100);comment:文件类型" json:"type"`
	URL        string `gorm:"column:url;type:text;comment:文件url" json:"url"`
	Name       string `gorm:"column:name;type:varchar(100);not null;comment:文件原始名" json:"name"`
	ObjectName string `gorm:"column:object_name;type:varchar(100);not null;comment:文件存储名" json:"objectName"`
	Size       string `gorm:"column:size;type:varchar(100);not null;default:'0';comment:文件大小" json:"size"`
	Path       string `gorm:"column:path;type:varchar(255);not null;comment:文件路径" json:"path"`
	MD5        string `gorm:"column:md5;type:char(32);not null;uniqueIndex:md5,md5_key;comment:文件的MD5值，用于比对文件是否相同" json:"md5"`
}
